Stratis.BitcoinD/Program.cs

```diff
@@ -21,46 +21,68 @@ public static void Main(string[] args)
 			Logs.Configure(new LoggerFactory().AddConsole(LogLevel.Trace, false));
 			NodeSettings nodeSettings = NodeSettings.FromArguments(args);
 
-			var node = (FullNode) new FullNodeBuilder()
-				.UseNodeSettings(nodeSettings)
-				.UseBlockStore()
-				.UseMempool()
-				.Build();
-
-			// == shout down thread ==
-			new Thread(() =>
-			{
-				Console.WriteLine("Press one key to stop");
-				Console.ReadLine();
-				node.Dispose();
-			})
+			try	
 			{
-				IsBackground = true //so the process terminate
-			}.Start();
+				// Verify folders are accessible by attempting to create sub directories,
+				// and removing them after the test.
+				var dataFolder = new DataFolder(nodeSettings);
+				foreach(var path in new []{ dataFolder.BlockPath, dataFolder.ChainPath, dataFolder.CoinViewPath }) {
```

Comment:
Do you mean Directory.Contains ... the issue is that you have to check write access because a regular user will be able to enumerate the directories no problem, but only fail during write access.  I thought about checking the current user identity, but that seems to not be available in .NET Core ... maybe I'm wrong.  So, I just tried writing a temp subdir to test write access.  Since this is just a startup task, it's not a perf issue.  

I will move the code into Checks.cs as suggested, then any other future checks could go in there.  Good point.